src/OrleansRuntime/GrainDirectory/RemoteGrainDirectory.cs

```diff
@@ -336,24 +75,24 @@ public Task RegisterManySingleActivation(List<ActivationAddress> addresses, int
                 if (curGen == tuple.Item2 || curGen == GrainInfo.NO_ETAG)
                 {
                     // the grain entry either does not exist in the local partition (curGen = -1) or has not been updated
-                    result.Add(new Tuple<GrainId, int, List<Tuple<SiloAddress, ActivationId>>>(tuple.Item1, curGen, null));
+                    result.Add(new Tuple<GrainId, int, List<ActivationAddress>>(tuple.Item1, curGen, null));
                 }
                 else
                 {
                     // the grain entry has been updated -- fetch and return its current version
-                    Tuple<List<Tuple<SiloAddress, ActivationId>>, int> lookupResult = partition.LookUpGrain(tuple.Item1);
+                    var lookupResult = await router.LookupAsync(tuple.Item1, 1);
```

Comment:
Good catch! I believe this change in semantics was accidental (not 100% sure how this came to be... perhaps this is a remnant of an earlier version of this branch that did indeed make some semantic changes). I will try to revert it back to the original semantics.